examples/http/cisco_meraki_channel_utilisation.js

```diff
@@ -0,0 +1,231 @@
+        { label: "Device", valueType: D.valueType.STRING },
+        { label: "Channel", valueType: D.valueType.STRING },
+        { label: "Channel Utilization", unit: "%", valueType: D.valueType.NUMBER },
+        { label: "Wifi Utilization",  unit: "%", valueType: D.valueType.NUMBER },
+        { label: "Non Wifi Utilization",  unit: "%", valueType: D.valueType.NUMBER },
+        { label: "Start Timestamp", valueType: D.valueType.DATETIME },
+        { label: "End Timestamp", valueType: D.valueType.DATETIME }
+    ]
+);
+
+/**
+ * Function to retrieve network information from the Meraki Dashboard API.
+ * @returns {Promise} A promise that resolves with an array of network information.
+ */
+function getNetworkInfo() {
+    var d = D.q.defer();
+    var config = {
+        url: "/api/v1/organizations/" + organizationId + "/networks",
+        protocol: "https",
+        headers: {
+            "Authorization": "Bearer " + apiKey,
+            "Content-Type": "application/json"
+        }
+    };
+    device.http.get(config, function(error, response, body) {   
+        if (error) {
+            console.error(error);
+            D.failure(D.errorType.GENERIC_ERROR);  
+        } else if (response.statusCode == 404) {
+            D.failure(D.errorType.RESOURCE_UNAVAILABLE);
+        } else if (response.statusCode === 401) {
+            console.error("Invalid API key");
+            D.failure(D.errorType.AUTHENTICATION_ERROR);
+        } else if (response.statusCode != 200) {
+            console.error(body);
+            D.failure(D.errorType.GENERIC_ERROR);
+        } else {
+            var data = JSON.parse(body);
+            var networksInfo = [];
+            if (networkId.toUpperCase() === "ALL") {
+                data.forEach(function(network) {
+                    networksInfo.push({ id: network.id, name: network.name });
+                });
+            } else {
+                var networkFound = false;
+                for (var i = 0; i < data.length; i++) {
+                    if (data[i].id === networkId) {
+                        networksInfo.push({ id: data[i].id, name: data[i].name });
+                        networkFound = true;
+                        break;
+                    }
+                }
+                if (!networkFound) {
+                    console.error("Network with specified ID not found.");
+                }
+            }
+            d.resolve(networksInfo);
+        }
+    });
+    return d.promise;
+}
+
+/**
+ * Function to retrieve channel utilization data for each network.
+ * @param {Array} networksInfo Array of network information.
+ * @returns {Promise} A promise that resolves with an array of channel utilization information
+ */
+function getChannelUtilization(networksInfo) {
+    var promises = networksInfo.map(function(network) {
+        var d = D.q.defer();
+        var config = {
+            url: "/api/v1/networks/" + network.id + "/networkHealth/channelUtilization",
```

Comment:
You should be using the timestamp params if you are only taking a record of the last entries per channel

Reply:
Moreover we will most likely need to only fetch the last 10 minutes (also make sure our sample period is the same) and add that as part of the documentation to explain that the data is actually in 10 min intervals.